sql/ir_generator.go

```diff
@@ -72,17 +72,17 @@ func generateTrainIR(slct *extendedSelect, connStr string) (*codegen.TrainIR, er
 	}, nil
 }
 
-func generateAttributeIR(attrs *attrs) ([]*codegen.Attribute, error) {
-	ret := []*codegen.Attribute{}
+func generateAttributeIR(attrs *attrs) (map[string]interface{}, error) {
+	ret := make(map[string]interface{})
 	for k, v := range *attrs {
 		resolvedExpr, err := parseExpression(v)
 		if err != nil {
 			return nil, err
 		}
-		ret = append(ret, &codegen.Attribute{
-			Key:   k,
-			Value: resolvedExpr,
-		})
+		if _, ok := ret[k]; ok {
+			return nil, fmt.Errorf("duplicate attribute: %v=%v", k, resolvedExpr)
```

Comment:
The parser already checks the attribute duplication at 

https://github.com/sql-machine-learning/sqlflow/blob/01730675a6505d1384393a04f647f2dc380a211e/sql/sql.y#L103-L111

It is safe to remove this check.

Reply:
Done

Reply:
Sorry, I meant to delete the check at `ir_generator.go` and keeps the check at `sql.y`.